opensearch/src/main/java/org/opensearch/sql/opensearch/client/OpenSearchRestClient.java

```diff
@@ -54,6 +58,36 @@ public Map<String, IndexMapping> getIndexMappings(String... indexExpression) {
     }
   }
 
+  @Override
+  public Map<String, Integer> getIndexMaxResultWindows(String... indexExpression) {
```

Comment:
do we need mapping or just single maxResultWindow?

Reply:
I approach this in a way similar to how `getIndexMappings` handles it.
For the client, `getIndexMappings` and `getIndexMaxResultWindows` return a Mapping from the index name to the corresponding result, without extra business logic.
In `OpenSearchDescribeIndexRequest`, `getMaxResultWindow` chooses the minimum of these values. This is specific to our need: if there's multiple indices, get the minimum max_result_window. Similarly, `getFieldTypes` unions the mappings of multiple indices here.